libs/promises/api/celix/DefaultScheduledExecutor.h

```diff
@@ -0,0 +1,138 @@
+/**
+ *Licensed to the Apache Software Foundation (ASF) under one
+ *or more contributor license agreements.  See the NOTICE file
+ *distributed with this work for additional information
+ *regarding copyright ownership.  The ASF licenses this file
+ *to you under the Apache License, Version 2.0 (the
+ *"License"); you may not use this file except in compliance
+ *with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *Unless required by applicable law or agreed to in writing,
+ *software distributed under the License is distributed on an
+ *"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ *specific language governing permissions and limitations
+ *under the License.
+ */
+
+#pragma once
+
+#include <memory>
+#include <future>
+#include <functional>
+#include <unordered_map>
+
+#include "celix/IScheduledExecutor.h"
+
+namespace celix {
+
+    class DefaultDelayedScheduledFuture : public celix::IScheduledFuture {
+    public:
+        explicit DefaultDelayedScheduledFuture(std::chrono::duration<double, std::milli> _delayInMs) : delayInMs{_delayInMs} {}
+
+        ~DefaultDelayedScheduledFuture() noexcept override = default;
+
+        bool isCancelled() const override {
+            std::lock_guard<std::mutex> lock{mutex};
+            return cancelled;
+        }
+
+        bool isDone() const override {
+            std::lock_guard<std::mutex> lock{mutex};
+            return done;
+        }
+
+        void setDone() {
+            std::lock_guard<std::mutex> lock{mutex};
+            done = true;
+            cond.notify_all();
+        }
+
+        void cancel() override {
+            std::lock_guard<std::mutex> lock{mutex};
+            cancelled = true;
+            cond.notify_all();
+        }
+
+        std::chrono::duration<double, std::milli> getDelayInMs() const {
+            std::lock_guard<std::mutex> lock{mutex};
+            return delayInMs;
+        }
+
+        void waitFor(std::chrono::duration<double, std::milli> time) {
+            std::unique_lock<std::mutex> lock{mutex};
+            cond.wait_for(lock, time);
+        }
+    private:
+        [[nodiscard]] std::chrono::duration<double, std::milli> getDelayOrPeriodInMilli() const override {
```

Comment:
Unused function?

Reply:
Yeah. Part of the Java ScheduledFuture interface, but unused in C++ so removed the functions for now.